src/arr/compiler/well-formed.arr

```diff
@@ -892,7 +896,10 @@ well-formed-visitor = A.default-iter-visitor.{
   end
 }
 
-top-level-visitor = A.default-iter-visitor.{
+top-level-visitor = well-formed-visitor.{
+  # Most behaviors are delegated to the non-toplevel visitor,
+  # but these expression forms are handled specially at the toplevel:
```

Comment:
I'm suspicious of this change, but can't point to the (regression) test that would fail because of it. Something about toplevel vs. non-toplevel checks about `data` is flashing in my memory.